schemas/alpha/azure.yaml.json

```diff
@@ -660,6 +673,69 @@
                     "description": "Optional. Used as the default environment type for this project."
                 }
             }
+        },
+        "workflow": {
+            "anyOf": [
+                {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "required": ["steps"],
+                    "properties": {
+                        "steps": {
+                            "type": "array",
+                            "title": "The steps to execute in the workflow",
+                            "description": "The steps to execute in the workflow. (Example: provision, package, deploy)",
+                            "minItems": 1,
+                            "items": {
+                                "type": "object",
+                                "$ref": "#/definitions/workflowStep"
+                            }
+                        }
+                    }
+                },
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "$ref": "#/definitions/workflowStep"
+                    }
+                }
+            ]
+        },
+        "workflowStep": {
+            "properties": {
+                "azd": {
+                    "title": "The azd command command configuration",
+                    "description": "The azd command configuration to execute. (Example: up)",
+                    "$ref": "#/definitions/azdCommand"
+                }
+            }
+        },
+        "azdCommand": {
+            "anyOf": [
+                {
+                    "type": "string",
+                    "title": "The azd command to execute",
+                    "description": "The name and args of the azd command to execute. (Example: deploy --all)"
+                },
+                {
+                    "type": "object",
```

Comment:
You kind of get both with this design.  If you take a look at the unit tests for workflow you can see it can be as simple as a string array or complex with the whole object.